### PR TITLE
Fold runtime data redacting into the runtime adapter.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java [hunk 1]
```diff
@@ -105,6 +105,12 @@ private RuntimeFieldBinding(int tag, String name, Message.Datatype datatype,
     this.messageField = messageField;
     this.builderField = builderField;
     this.builderMethod = builderMethod;
+
+    if (redacted && label == Message.Label.REQUIRED) {
```
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java]: This check should be in the compiler.

Comment (swankjesse) [inline wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java]: There’s 2 reasons to use redacted:

 * To obscure values in `toString()`
 * To strip fields

Presumably redacted+required is safe for the first case?

File: wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java [hunk 2]
```diff
@@ -152,6 +158,23 @@ void setBuilderField(Object builder, Object value) {
     }
   }
 
+  public void redactBuilderField(Message.Builder<?> builder) {
+    if (!redacted && datatype != Message.Datatype.MESSAGE) return;
```
Comment (swankjesse) [inline wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java]: not sure I follow this: who redacts scalar fields?

Comment (swankjesse) [inline wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java]: Ahh. . . it's polymorphic. Duh.

**Comment by swankjesse** [comment]
LGTM

File: wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java (hunk 1 above)
Comment (JakeWharton) [inline wire-runtime/src/main/java/com/squareup/wire/RuntimeFieldBinding.java]: Ah, good catch. Will move to `redact`.

**Pull request merged**